src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -329,6 +364,28 @@ public void ApplyForwarders(HttpContext context)
                 }
                 request.Host = HostString.FromUriComponent(currentValues.Host);
             }
+
+            if (checkPrefix && currentValues.Prefix != null)
+            {
+                if (request.PathBase.HasValue)
+                {
+                    // Save the original
+                    requestHeaders[_options.OriginalPrefixHeaderName] = request.PathBase.ToString();
+                }
+
+                if (forwardedPrefix!.Length > entriesConsumed)
+                {
+                    // Truncate the consumed header values
+                    requestHeaders[_options.ForwardedPrefixHeaderName] = forwardedPrefix.Take(forwardedPrefix.Length - entriesConsumed).ToArray();
```

Comment:
Why is this using LINQ?  Let's aim to avoid *all* temporary allocations.

Reply:
> Why is this using LINQ?

Just followed the existing pattern in the class.

Have pushed new commits that change all the cases where header values are truncated to use a new helper method that uses Array.Copy instead of LINQ.